File: Data Structures & Algorithms/eating-bananas/submission-2.py
class Solution:
    def minEatingSpeed(self, piles: List[int], h: int) -> int:
        l, r = 1, max(piles)
        res = r

        while (l <= r):
            mid = (l+r) // 2
            hour = self.checkHourK(piles, mid)
            if hour <= h:
                res = mid
                r = mid - 1
            else:
                l = mid + 1
        return res

    def checkHourK(self, piles: List[int], h: int) -> int:
        sum = 0
        for i in piles:
            sum = sum + math.ceil(float(i)/h)
        return sum

File: Data Structures & Algorithms/pow-x-n/submission-2.java
class Solution {
    public double myPow(double x, int n) {
        if (n == 0) {
            return 1;
        }
        if (n == 1) {
            return x;
        }
        if (n < 0) {
            return 1 / myPow(x, -n);
        } else {
            if (n % 2 == 1) {
                return x * myPow(x*x, n/2);
            } else {
                return myPow(x*x, n/2);
            }
        }
    }
}
